src/NumSharp.Core/Selection/NDArray.Indexing.Masking.cs

```diff
@@ -23,7 +23,11 @@ public partial class NDArray
             get => FetchIndices(this, np.nonzero(mask), null, true);
             set
             {
-                throw new NotImplementedException("Setter is not implemented yet");
+                for (int i = 0; i < mask.size; i++)
```

Comment:
It won't work on `mask.ndim>1 dimensions` because GetBoolean takes int[].
Please take a look [here](https://docs.scipy.org/doc/numpy-1.17.0/user/basics.indexing.html#index-arrays), the algorithm at minimum must be compliant to the behavior described by scipy.

Reply:
Acturally, I have not idea how to do support any dimemsion.